src/hb-ot-stat-table.hh

```diff
@@ -230,18 +252,58 @@ struct STAT
   bool sanitize (hb_sanitize_context_t *c) const
   {
     TRACE_SANITIZE (this);
-    return_trace (likely (c->check_struct (this) &&
-			  majorVersion == 1 &&
-			  minorVersion > 0 &&
+    return_trace (likely (version.sanitize (c) &&
+                          c->check_struct (this) &&
+			  version.major == 1 &&
 			  designAxesOffset.sanitize (c, this, designAxisCount) &&
 			  offsetToAxisValueOffsets.sanitize (c, this, axisValueCount, &(this+offsetToAxisValueOffsets))));
   }
+  
+  bool has_data () const { return version.to_int (); }
+
+  unsigned get_design_axis_count () const { return designAxisCount; }
+
+  unsigned get_axis_record_name_id (unsigned int axis_record_index) const
+  {
+    if (unlikely (axis_record_index >= designAxisCount)) return HB_OT_NAME_ID_INVALID;
+    const StatAxisRecord &axis_record = get_design_axes ()[axis_record_index];
+    return axis_record.get_name_id ();
+  }
+
+  unsigned get_axis_value_count () const { return axisValueCount; }
+
+  unsigned get_axis_value_name_id (unsigned int axis_value_index) const
+  {
+    if (unlikely (axis_value_index >= axisValueCount)) return HB_OT_NAME_ID_INVALID;
+    const AxisValue &axis_value = (this + get_axis_value_offsets ()[axis_value_index]);
+    return axis_value.get_value_name_id ();
+  }
+
+  void collect_name_ids (hb_set_t *nameids_to_retain) const
+  {
+    if (!has_data ()) return;
+
+    + get_design_axes ()
+    | hb_map (&StatAxisRecord::get_name_id)
+    | hb_sink (nameids_to_retain)
+    ;
+
+    + get_axis_value_offsets ()
+    | hb_apply ([&] (const OffsetTo<AxisValue>& _) { hb_set_add (nameids_to_retain, (this + _).get_value_name_id ()); })
```

Comment:
Move the `hb_set_add` to be a `hb_sink` still.  The `get_value_name_id` also can be a `hb_map` by itself.  You just need a lambda to add `this` to the offset.

I'll probably add a method to do that without having to define a lambda.